internal/dispatch/keys/dispatchcheckkey.go

```diff
@@ -0,0 +1,27 @@
+package keys
+
+import (
+	"encoding/binary"
+)
+
+// DispatchCacheKey is a struct which holds the key representing a dispatch operation being
+// dispatched or cached. This key is guaranteed to be unique in byte or (uint64, uint64) form for
```

Comment:
I don't like the comment. It's not actually guaranteed to be unique, the only way to guarantee that would be to use something like a perfect hash or track all keys and check for collisions.

Reply:
Moved and changed